src/main/template/cassandra.yaml

```diff
@@ -0,0 +1,1064 @@
+              mode: 000644
+        install_cassandra:
+          packages:
+            yum:
+              cassandra: []
+        configure_cassandra:
+          commands:
+            01_configure_hosts:
+              command: !FindInMap ["Scripts", "Cassandra", "CreateHostsEntry"]
+            02_setup_data_directories:
+              command: !FindInMap ["Scripts", "Cassandra", "SetupDataDirectories"]
+            03_update_cassandra_yaml:
+              command: !FindInMap ["Scripts", "Cassandra", "UpdateCassandraYaml"]
+              env:
+                CLUSTER_NAME: !Ref clusterName
+                SEED_NODES: !Join
+                  - ""
+                  - - "a.seed."
+                    - !Ref internalDomain
+                    - ",b.seed."
+                    - !Ref internalDomain
+                    - ",c.seed."
+                    - !Ref internalDomain
+            04_update_cassandra_env_sh:
+              command: !FindInMap ["Scripts", "Cassandra", "UpdateCassandraEnvSh"]
+            05_update_jvm_options:
+              command: !FindInMap ["Scripts", "Cassandra", "UpdateJvmOptions"]
+            06_update_cassandra_rackdc_properties:
+              command: !FindInMap ["Scripts", "Cassandra", "UpdateCassandraRackdcProperties"]
+        start_cassandra:
+          services:
+            sysvinit:
+              cassandra:
+                enabled: true
+                ensureRunning: true
+        ensure_cassandra_is_running:
+          commands:
+            01_sleep:
+              command: sleep 60
+            02_check:
+              command: nodetool info
+
+  autoScalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    DependsOn:
+      - seedADnsRecord
+      - seedBDnsRecord
+      - seedCDnsRecord
+    Properties:
+      LaunchConfigurationName: !Ref launchConfiguration
+      MaxSize: 30
+      MinSize: 0
+      DesiredCapacity: 1
+      VPCZoneIdentifier:
+        - Fn::Select:
+          - 0
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+        - Fn::Select:
+          - 1
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+        - Fn::Select:
+          - 2
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} AZ
+          PropagateAtLaunch: true
+
+Outputs:
+  securityGroup:
+    Description: The cassandra security group, so that other security groups can allow SSH ingress from it, or register
+      additional security group ingress on it.
+    Value: !Ref securityGroup
+    Export:
+      Name: !Sub ${AWS::StackName}-security-group
+
+  seedNodes:
+    Description: The list of seed nodes to be used by connecting systems.
+    Value: !Join
+     - ""
+     - - "a.seed."
```

Comment:
How about using `!Ref seedADnsRecord` instead? [AWS::Route53::RecordSet -> Return Values -> Ref](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-route53-recordset.html#aws-properties-route53-recordset-return-values) states that
> When you pass the logical ID of this resource to the intrinsic Ref function, Ref returns the value of the domain name of the record set, such as acme.example.com.

This syntax would create a real relationship between the seed list and the domain entries as opposed to a loose coupling achieved by correct string concatenation.